server/lib/pbench/test/unit/test_process_tb.py

```diff
@@ -30,36 +49,57 @@ def mock_get(section: str, option: str):
 
     def test_get_receive_dir_value_failed(self, monkeypatch, test_logger):
         def mock_get(section: str, option: str):
-            assert option == "pbench-receive-dir-prefix"
+            assert (
+                option == "pbench-receive-dir-prefix"
+            ), f"Unexpected value in option, {section!r}/{option!r}, in mocked config.get()"
             return fixture_path / "wrong-directory"
 
-        expected_error_path = str(fixture_path / "wrong-directory-002")
+        error_msg = "does not exist, or is not a directory] "
+        expected_error_msg = error_msg + str(
+            fixture_path.resolve() / "wrong-directory-002"
+        )
+
         monkeypatch.setattr(MockConfig, "get", mock_get)
-        with pytest.raises(Exception) as e:
+        with pytest.raises(NotADirectoryError) as e:
             ProcessTb(MockConfig, test_logger)
 
-        assert expected_error_path in str(e)
+        assert expected_error_msg in str(e)
 
     def test_process_tb_file_not_found_error(self, monkeypatch, caplog, test_logger):
         """checks normal processing when tar ball is not present"""
-        tb_path = fixture_path / "bad_tarball"
+        tb_path = fixture_path
+        receive_path = Path(str(tb_path.resolve()) + "-002")
+        bad_tb = receive_path / bad_tarball
 
         def mock_get(self, section: str, option: str):
-            assert option in ["pbench-receive-dir-prefix", "put-token"]
+            assert option in [
+                "pbench-receive-dir-prefix",
+                "put-token",
+            ], f"Unexpected configuration option, {section!r}/{option!r}, in mocked config.get()"
             return (
                 tb_path
                 if option == "pbench-receive-dir-prefix"
                 else "Authorization-token"
             )
 
+        def mock_glob(path, pattern):
+            tbmd5 = path / "bad_log.tar.xz.md5"
+            if path == receive_path and pattern == "**/*.tar.xz.md5":
+                assert (
+                    tbmd5.suffix == ".md5"
+                ), f"Unexpected tar ball, {path!r} in mocked Path.glob()"
+                return [tbmd5]
```

Comment:
There's a way out of this function which does not raise an exception and does not return a value...which is Bad.

I think we can plug that leak by reworking this in a more straightforward fashion:
```
        def mock_glob(path, pattern):
            assert path == receive_path and pattern == "**/*.tar.xz.md5"
            return [receive_path / "bad_log.tar.xz.md5"]
```
We can predict what `path` and `pattern` should be, so we check those, and then we return the value that we want the CUT to receive.  (I dropped the local variable since it's used in only one place and the expression it refers to is so simple.)

One other note:  in your code above, we _know_ what value `tbmd5` has (because we assigned it!) so the assertion that the suffix is `.md5` will never fail (so what's the point?).  We _could_ assert that the value that we're intending to return matches the requested pattern, but, since we know the pattern in advance and since we control the value being returned, I think we can count on the person writing the code to make sure that they match.  (And, since the value being returned doesn't change and since the test code confirms that the pattern is what we expect, the test should detect the problem if value being returned stops matching.)

Reply:
The same comment applies to `mock_glob()` in `test_process_tb_connection_error()` and probably the others.  (Except for `test_process_tb_zero()` -- you're already doing it there!)

Reply:
By the way, I would recommend a slightly different signature for `mock_glob()`.  Since it's replacing an instance method on a `Path` object, it should be:
```
        def mock_glob(self: Path, pattern: str) -> List[Path]:
```
The first parameter in an instance method _can_ be named anything, but it is conventionally named `self`.  And, the value returned really should be a list of `Path`'s, although, apparently, we are getting away with returning strings (that's a credit either to the completeness of your mocking or to the looseness of Python's typing model 😉).